### PR TITLE
Similar request error handling fix to #167 for remaining requests

### PR DESCRIPTION
**Pull request by markberger**
I am getting `JSONDecodeError` messages when receiving a 504 response in `def _get` [here](https://github.com/VingtCinq/python-mailchimp/blob/master/mailchimp3/mailchimpclient.py#L163) similar to #167. This implements the same fix that is in 3377ea9 and 95f1edc for other requests.

### CODE REVIEW

**Comment by stephenross** [comment]
That looks good, update the docs with the information if it needs it and increment the version numbers in the readme files and the setup.py. Also, if you could provide any info you have on which specific calls you are making that are returning 504 in a comment on this, that would also be helpful.

**Comment by markberger** [comment]
The 504 error occurs intermittently when I call `client.reports.email_activity.all`. After reaching out to mailchimp support, they said that hitting this endpoint with `count=1000` is very expensive on their end. They suggested that I reduce the count to 500. Unfortunately the count of 1000 is hardcoded here:

https://github.com/VingtCinq/python-mailchimp/blob/master/mailchimp3/baseapi.py#L36

I'll open a second pr that drops this hardcoded value.

**Comment by markberger** [comment]
@stephenross this should now be good to go

File: README.rst [hunk 1]
```diff
@@ -928,4 +928,3 @@ The project is licensed under the MIT License.
    :target: https://pypi.python.org/pypi/mailchimp3
 .. |MIT license| image:: https://img.shields.io/badge/licence-MIT-blue.svg
 .. |Stable| image:: https://img.shields.io/badge/status-stable-green.svg
-
```
Comment (stephenross) [inline README.rst]: Please leave the trailing newline in the file.

**Comment by stephenross** [comment]
@charlesthk If you could put this on pypi, that would be awesome

**Comment by charlesthk** [comment]
@stephenross v3.0.9 is now on Pypi ! Thanks.

**Pull request merged**